app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryModel.kt

```diff
@@ -11,23 +11,29 @@ data class CustomDrugEntryModel(
     val dosage: String?,
     val dosageHasFocus: Boolean?,
     val frequency: DrugFrequency?,
-    val rxNormCode: String?
+    val rxNormCode: String?,
+    val dosagePlaceholder: String
 ) : Parcelable {
   companion object {
     fun default(
-        openAs: OpenAs
+        openAs: OpenAs,
+        dosagePlaceholder: String
     ) = CustomDrugEntryModel(
         openAs = openAs,
         drugName = null,
         dosage = null,
         dosageHasFocus = null,
         frequency = null,
-        rxNormCode = null)
+        rxNormCode = null,
+        dosagePlaceholder = dosagePlaceholder)
   }
 
   val isDrugFrequencyUnknown
     get() = frequency is DrugFrequency.Unknown
 
+  val isDosageEqualToPlaceHolderOrEmpty
```

Comment:
We should not be doing this to track the placeholder content. We can switch to using suffix feature of the text field ([refer](https://obvious-hq.slack.com/archives/CE0F1TLTC/p1629780370039200)).

But in general, we need to track the text field focus and if a text field has text or not to set the place holder. Please refer how `CustomPrescriptionEntryUiRenderer` does this.

Reply:
It seems like we are doing this to set empty string if there is only `mg` unit? It is not how it should be handled.

Reply:
✅ But in general, we need to track the text field focus and if a text field has text or not to set the place holder. Please refer how CustomPrescriptionEntryUiRenderer does this.

As discussed offline going with the previous approach of showing `mg` once the user has focused on the dosage edit text and pressed saved even if there is no numeric value added by the user for dosage.